### PR TITLE
Update `BEGIN` doc to clarify isolation levels

### PR DESCRIPTION
**Pull request by chuck-alt-delete**
Currently, the doc for [BEGIN](https://materialize.com/docs/sql/begin/) says:
> ISOLATION LEVEL SERIALIZABLE | A no-op because transactions are always serializable. Lower isolation modes are also accepted, but treated identically to serializable.

I believe this is outdated. The default isolation level is strict serializable. Also, it was confusing (to me) that `READ ONLY` was being talked about in the same variable as isolation level. I updated the diagram:

image
 
as well as the text (see file changes)

### CODE REVIEW

File: doc/user/sql-grammar/sql-grammar.bnf [hunk 1]
```diff
@@ -16,7 +16,7 @@ array_agg ::=
 as_of ::=
   'AS' 'OF' ( 'AT' 'LEAST' )? timestamp_expression
 begin ::=
-  'BEGIN' ( transaction_mode ( ',' transaction_mode )* )?
+  'BEGIN' ( 'ISOLATION LEVEL' ( isolation_level) | ('READ ONLY')?)
```
Comment (mjibson) [inline doc/user/sql-grammar/sql-grammar.bnf:19]: This isn't correct, you can specify both isolation level and read only. I modeled this page after the postgres page which groups these options together: https://www.postgresql.org/docs/current/sql-begin.html

I think this diff chunk should be reverted.

File: doc/user/content/sql/begin.md [hunk 2]
```diff
@@ -27,7 +26,7 @@ Transactions in Materialize do not support interleaving arbitrary kinds of state
 
 ### Read-only transactions
 
-A **read-only** transaction starts with a [`SELECT`](/sql/select) statement and allows only `SELECT` statements.
```
Comment (mjibson) [inline doc/user/content/sql/begin.md:30]: The `READ ONLY` syntax and a "Read-only transaction" are slightly different concepts and I think the current wording is more accurate than this diff. All txns in mz are either read-only, write-only, or "a single SUBSCRIBE". We decide which txn type based on the first statement a user attempts to execute. The `READ ONLY` syntax prevents a write-only txn from starting.

I think this diff chunk should be reverted.

File: doc/user/content/sql/begin.md [hunk 3]
```diff
@@ -11,12 +11,11 @@ menu:
...
 {{< diagram "begin.svg" >}}
 
-Supported `transaction_mode` option values:
+
 
 Value | Description
 ------|----------
-`ISOLATION LEVEL SERIALIZABLE` | A no-op because transactions are always serializable. Lower isolation modes are also accepted, but treated identically to serializable.
-`READ ONLY` | Limits the transaction to read-only operations.
+`isolation_level` | The default isolation level is `STRICT SERIALIZABLE`. You can also use `SERIALIZABLE`. While any isolation level below `SERIALIZABLE` is also accepted as input, the isolation level will always be at least `SERIALIZABLE`. See [consistency guarantees](/overview/isolation-level) for more details.
```
Comment (mjibson) [inline doc/user/content/sql/begin.md:18]:
```suggestion
`ISOLATION LEVEL <level>` | Sets the transaction [isolation level](/overview/isolation-level).
`READ ONLY` | Limits the transaction to read-only operations.
```

I think we should take the same tack as postgres, and only document these in a single place.

File: doc/user/sql-grammar/sql-grammar.bnf (hunk 1 above)
Comment (chuck-alt-delete) [reply doc/user/sql-grammar/sql-grammar.bnf]: Done

File: doc/user/content/sql/begin.md (hunk 2 above)
Comment (chuck-alt-delete) [reply doc/user/content/sql/begin.md]: Done

File: doc/user/content/sql/begin.md [hunk 4]
```diff
@@ -11,11 +11,11 @@ menu:
 
 {{< diagram "begin.svg" >}}
 
-Supported `transaction_mode` option values:
```
Comment (mjibson) [inline doc/user/content/sql/begin.md:14]: I think this needs to stay too

Comment (chuck-alt-delete) [reply doc/user/content/sql/begin.md]: D’oh! Yes. Ok done.

**Review by mjibson — APPROVED** [review]

**Pull request merged**